spatial/r3/triangle.go

```diff
@@ -0,0 +1,72 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3
+
+import "math"
+
+// Triangle represents a triangle in 3D space and
+// is composed by 3 vectors corresponding to the position
+// of each of the vertices. Ordering of these vertices
+// decides the "normal" direction.
+type Triangle struct {
+	// Triangle vertices. Ordering of these vertices decides Normal vector direction.
+	// Inverting ordering of two vertices inverts the resulting direction.
+	V [3]Vec
+}
+
+// Normal returns the vector with direction
+// perpendicular to the Triangle's face. The ordering
+// of the triangle vertices decides the normal's resulting
+// direction. Normal is not guaranteed to return a unit-norm vector.
+func (t Triangle) Normal() Vec {
+	s1, s2, _ := t.sides()
```

Comment:
it seems (moderately) wasteful to ask for the computation of `Sub(t.V[0], t.V[2])` to drop it right there.
but ok.

Reply:
Would the SSA not optimize the calculation? Maybe not today but it does seem like something that could done given it looks like a short enough path